src/coreComponents/physicsSolvers/SolverBase.hpp

```diff
@@ -666,6 +666,46 @@ class SolverBase : public ExecutableGroup
     }
   }
 
+  template< typename LAMBDA >
+  void forMeshTargets( std::vector< const MeshBody * > const & meshBodies, LAMBDA && lambda ) const
+  {
+    for( auto const & target: m_meshTargets )
+    {
+      string const meshBodyName = target.first;
+      arrayView1d< string const > const & regionNames = target.second.toViewConst();
+      for( const MeshBody * meshBody: meshBodies )
+      {
+        if( meshBodyName != meshBody->getName() )
+        { continue; }
+
+        for( MeshLevel const * meshLevel: meshBody->getMeshLevels() )
+        {
+          lambda( meshBodyName, *meshLevel, regionNames );
+        }
+      }
```

Comment:
This impl is not very good, but if we go that way, other accessors would be done.
Maybe we can keep it also since we'll never have huge number of `MeshBody`.

Reply:
I think we can change `m_meshTargets` s.t. we don't need to pass `meshBodies`